vault/resource_aws_auth_backend_client_test.go

```diff
@@ -240,13 +241,15 @@ func testAccAWSAuthBackendClientCheck_attrs(backend string) resource.TestCheckFu
 			consts.FieldSTSEndpoint:            consts.FieldSTSEndpoint,
 			consts.FieldSTSRegion:              consts.FieldSTSRegion,
 			consts.FieldIAMServerIDHeaderValue: consts.FieldIAMServerIDHeaderValue,
+			consts.FieldMaxRetries:             consts.FieldMaxRetries,
 		}
 		for stateAttr, apiAttr := range attrs {
-			if resp.Data[apiAttr] == nil && instanceState.Attributes[stateAttr] == "" {
-				continue
+			respApiAttr := resp.Data[apiAttr]
+			if apiAttr == consts.FieldMaxRetries {
+				respApiAttr = resp.Data[apiAttr].(json.Number).String()
```

Comment:
Can we add a nil check here? I recently ran into a bit of code like this, and Vault's response behavior had changed for the test to return `nil` for that value which caused a panic on lines like this. 

We should add a nil check for `resp.Data[appiAttr]` that returns an error if the value is nil and was expected. If it is non nil we should be able to convert it to a json.Number

Reply:
I am sorry, I merged it right before you put up this comment. I opened up a new PR to address this: https://github.com/hashicorp/terraform-provider-vault/pull/2275

Reply:
Good catch, do you think both `resp.Data != nil` and `resp.Data[apiAttr] != nil` would need to be checked?

Reply:
@Zlaticanin no problem at all, thanks for putting up the follow up PR! 🙏🏼 

@robmonte, I think Vault's server returning `nil` for `resp.Data` in this case would be very odd behavior, I think panicking on that might be better? I think just the `resp.Data[apiAttr] != nil` check should suffice